### DISCUSSION TITLE
Change a project's name and/or URL independently of each other

### DISCUSSION THREAD

**Discussion by greystate** [umbraco/Umbraco.Cloud.Issues#1010] [Ideas]
I'd love to be able to rename a project without changing the project's URL, and vice versa - e.g., if I create a project with the name **Spørg Før Du Går**, it would the weird URL `sp-rg-f-r-du-g-r.s1.umbraco.io`. Here, I'd like to be able to change the URL part to something simple, like `spdg` or `spoerg`.

On the other hand, I could have named the project **Spoerg Foer Du Gaar** to get the URL `spoerg-foer-du-gaar.s1.umbraco.io` - but then I'd have a not so great projectname to look at every time I log on to the portal.
 
As it is now, I have to pick which one I want to be "correct" and then just accept that the other one isn't so great. Catch 22.

There was a similar request on the old tracker as [UAAS-478](https://issues.umbraco.org/issue/UAAS-478).

**Comment by nul800sebastiaan** [comment]
Thanks @greystate - I'll have to check with the Cloud team as to how this all works, I'll get back to you.

**Comment by nul800sebastiaan** [comment]
Hi @greystate - I had a chat with them and this request makes sense. We don't have immediate plans to work on this but we have put this on the ideas list.

It seems like the easiest thing to do is to give you an automatically generated URL to begin with (the ugly one) and then you can decide to rename the project URL afterwards. We should probably also just have the option to rename the display name of the project as Bjarne asked for in the related issue you linked.

**Comment by azure-devops-sync** [comment]
AB#655

**Comment by greystate** [comment]
Just had a thought about this:

***
Would it possible to update the `/environments/@name` attribute in the `UmbracoDeploy.config` file and have that be the display name in the portal, or are they not connected at all?
***

I could easily live with that; thus, creating the project with the "best alias" method and subsequently editing the .config to my liking. Even looks like the color is in there too...

**Comment by sitereactor** [comment]
@greystate the two are not connected. The values in the UmbracoDeploy.config file is static values, which are essentially read-only. So they are just used to display the correct names, urls, etc. in the Deploy dashboard in the Backoffice.

**Comment by greystate** [comment]
Thanks @sitereactor — didn't really think _I should be so lucky_ either :)

**Comment by sajumb** [comment]
We have created a ticket for this in the product backlog. It is a candidate to be included in the upcoming sprints.

**Comment by greystate** [comment]
Woohoo - Christmas Present!!
😄

**Comment by greystate** [comment]
Just to let everyone know, I've added this to my “Christmas <del>2021</del> <del>2022</del> <del>2023</del> <ins>2024</ins>” wishlist 😬

**Comment by bjarnef** [comment]
I often comes back to this issue and often we have some long project names, where we would like to keep the alias short and simple.
There are also some default culture specific replacement, but it may not be the desired replacement, but unlike in the CMS with URL replacement, it isn't possible to override this.

E.g. as replacement to Danish letter `ø` some may prefer `o` or `oe`.

I think it would be great with some e.g. input with default alias generated, but the option with unlock to edit the alias.
Eventually also the option to edit alias afterwards.

Furthermore maybe there shouldn't be any limit in have many times the project can be renamed? Besides that a description could be useful the add a bit more detail of the project.

**Comment by greystate** [comment]
Just to state my position here:

I'd be totally fine with not being able to change the alias once that's locked in - but I would *really* like to be able to change the name that's displayed.

**Comment by greystate** [comment]
Who's in for a **Kickstarter** campaign to fund the hours of implementing this feature?

(_Just kidding,_ of course 😢 🐼)

**Comment by bjarnef** [comment]
It would be great if it shows generated alias next to input similar to locked alias in CMS.
Then we can adjust this while creating the project,

and then move "Name is available" + checkbox below input field.

<img width="1084" height="273" alt="Image" src="https://github.com/user-attachments/assets/adf07586-d8cf-4c78-a228-5a02076ef3a1" />

Currently one has to name the project first to get a "nice" alias and then rename project later.
However there is a technical limitation (I have been told), so alias must be minimum 4 characters.

Anyway adjusting alias would be useful as if may not generate an ideal alias depending on culture.

IIRC it was previously possible to rename a project afterwards without affecting the alias.
It seems no longer to be the case.

Renaming a project will now also update the alias and the Umbraco Cloud URLs.

It is common that projects/companies are rebranded, but not necessary a requirement to update/change alias.
Often we also just want to rename a project to e.g. include a short name or abbreviation.
